rslib/src/storage/deck/mod.rs

```diff
@@ -198,13 +198,15 @@ impl SqliteStorage {
             .collect()
     }
 
-    pub(crate) fn child_decks(&self, parent: &Deck) -> Result<Vec<Deck>> {
+    pub(crate) fn child_decks(&self, parent: &Deck, sorted: bool) -> Result<Vec<Deck>> {
         let prefix_start = format!("{}\x1f", parent.name);
         let prefix_end = format!("{}\x20", parent.name);
         self.db
-            .prepare_cached(concat!(
+            .prepare_cached(&format!(
+                "{}{}{}",
                 include_str!("get_deck.sql"),
-                " where name >= ? and name < ?"
+                " where name >= ? and name < ?",
+                if sorted { " order by name" } else { "" }
```

Comment:
That ordering should be free due to the use of the index, and I'd expect you'd get it even if you didn't explicitly request it:

```
>>> pp(mw.col.db.all("explain query plan select id, name from decks where name >= 'foo' and name < 'foo1' order by name"))
[[3,
  0,
  0,
  'SEARCH decks USING COVERING INDEX idx_decks_name (name>? AND name<?)']]

>>> pp(mw.col.db.all("explain query plan select id, name from decks where name >= 'foo' and name < 'foo1'"))
[[2,
  0,
  0,
  'SEARCH decks USING COVERING INDEX idx_decks_name (name>? AND name<?)']]
```

It's good to be explicit about our intentions here to avoid a repeat of 9daa0c9acb3019c719a282c81888707b465bc9ce, but since this order should be free, I think we can probably omit the boolean and just always sort that way.

Reply:
> I'd expect you'd get it even if you didn't explicitly request it:

This doesn't seem to be always the case. I only made that change because the tree was malformed for certain decks.
But no objections to always return a sorted list, which matches the bahaviour of `get_all_deck_names()`.